sections/design/class-diagram.tex

```diff
@@ -0,0 +1,57 @@
+\section{Class diagram}
+A class diagram helps to create an understanding of the design of the system.
+The following diagram illustrates the relevant classes related to the problem domain for the system.
+It illustrates structural relations between classes and objects\cite{OOAD}.
+
+\begin{figure}[H]
+    \includegraphics[width=\linewidth]{/tutorclassdiagram.png}
+     \caption{The class diagram describing classes in the problem domain}
+     \label{fig:class-diagram}
+ \end{figure}
+
+\subsection{Users}
+Users are defined through the \texttt{users} cluster.
+A user aggregates 0 or more roles.
+A role is a generalization of either the tutor or administrator classes.
+The role class ensures that the user can have both of the different roles, only one role or neither role, and that it can be dynamically changed.
+If a user does not have a role, the user is simply regarded as a user of the system - a person seeking a tutor.
+Reports are used to ensure that the system is trustworthy and prevent the same problems recurring in interactions between students and tutors.
+A user is associated with reports through an association relation.
+A user will be associated with 0 or more reports, depending on how many times the student has reported reported a tutor.
+A tutor is also associated with 0 or more reviews in the same way.
+Students can rate their interactions with a tutor through the review.
```

Comment:
Synes sgu at mange af dem er sådan lidt "A user.. A user.. A tutor.. A user.. A role.. igennem hele den PR her, det er lidt nedern at læse

Reply:
Har lavet lidt ændringer, hvis det stadig er nedern får jeg nok brug for lidt eksempler på hvordan det skal ændres 🤔